### PR TITLE
Feature/alcs 776

### PR DESCRIPTION
**Pull request by mhuseinov**
- replace jsonb with table for lots on component
- create a new table to handle plan numbers for condition<->component
- inline editing for conditions lot
- inline editing for the condition to component plan numbers
- table comments

### CODE REVIEW

File: alcs-frontend/src/app/features/application/decision/conditions/condition/condition.component.ts [hunk 1]
```diff
@@ -31,16 +44,75 @@ export class ConditionComponent implements OnInit, AfterViewInit {
...
+          )
+          .sort((l1, l2) => l1.number - l2.number);
+        this.subdComponent = subdComponent;
+      }
+    }
+  }
+
+  async loadPlanNumber() {
+    const subdComponent = this.condition.components?.find(
+      (e) => e.applicationDecisionComponentTypeCode === APPLICATION_DECISION_COMPONENT_TYPE.SUBD
```
Comment (dhaselhan) [inline alcs-frontend/src/app/features/application/decision/conditions/condition/condition.component.ts:91]: e? component doesn't even start with the letter e. Rename to component.

File: alcs-frontend/src/app/features/application/decision/conditions/condition/condition.component.ts [hunk 2]
```diff
@@ -31,16 +44,75 @@ export class ConditionComponent implements OnInit, AfterViewInit {
...
+      ) &&
+      this.isRequireSurveyPlan
+    ) {
+      const planNumbers = (await this.conditionService.fetchPlanNumbers(this.condition.uuid)).filter(
+        (e) => e.applicationDecisionComponentUuid !== subdComponent?.uuid
+      );
+
+      this.planNumbers =
+        this.condition.components
+          ?.filter((e) => e.applicationDecisionComponentTypeCode !== APPLICATION_DECISION_COMPONENT_TYPE.SUBD)
```
Comment (dhaselhan) [inline alcs-frontend/src/app/features/application/decision/conditions/condition/condition.component.ts:106]: Another e, can you call this component?

File: alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-components/decision-components.component.ts [hunk 3]
```diff
@@ -116,7 +117,20 @@ export class DecisionComponentsComponent implements OnInit, OnDestroy, AfterView
           applicationDecisionComponentType: this.decisionComponentTypes.find(
             (e) => e.code === typeCode && e.uiCode !== 'COPY'
           ),
-          subdApprovedLots: this.application.submittedApplication?.subdProposedLots,
+          subdApprovedLots: this.application.submittedApplication?.subdProposedLots.map(
+            (e, index) =>
+              ({
+                ...e,
+                number: index,
+              } as ProposedDecisionLotDto)
```
Comment (dhaselhan) [inline alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-components/decision-components.component.ts:125]: Why as? Can we do a cleaner way than casting?

File: alcs-frontend/src/app/shared/inline-editors/inline-textarea/inline-textarea.component.spec.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { InlineTextareaComponent } from './inline-textarea.component';
 
-describe('InlineTextareaComponent', () => {
+describe('InlineTextComponent', () => {
```
Comment (dhaselhan) [inline alcs-frontend/src/app/shared/inline-editors/inline-textarea/inline-textarea.component.spec.ts:6]: I think this should stay as InlineTextareaComponent ?

File: services/apps/alcs/src/alcs/application-decision/application-component-lot/application-decision-component-lot.entity.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  constructor(data?: Partial<ApplicationDecisionComponentLot>) {
+    super();
+    if (data) {
+      Object.assign(this, data);
+    }
+  }
+
+  @AutoMap()
+  @Column({ type: 'int' })
+  number: number;
```
Comment (dhaselhan) [inline services/apps/alcs/src/alcs/application-decision/application-component-lot/application-decision-component-lot.entity.ts:28]: number is not very descriptive, maybe index or rank?

File: alcs-frontend/src/app/shared/inline-editors/inline-textarea/inline-textarea.component.spec.ts (hunk 4 above)
Comment (mhuseinov) [reply alcs-frontend/src/app/shared/inline-editors/inline-textarea/inline-textarea.component.spec.ts]: done

File: alcs-frontend/src/app/features/application/decision/conditions/condition/condition.component.ts (hunk 1 above)
Comment (mhuseinov) [reply alcs-frontend/src/app/features/application/decision/conditions/condition/condition.component.ts]: done. I think I've got all e, i and l that were added

File: services/apps/alcs/src/alcs/application-decision/application-component-lot/application-decision-component-lot.entity.ts (hunk 5 above)
Comment (mhuseinov) [reply services/apps/alcs/src/alcs/application-decision/application-component-lot/application-decision-component-lot.entity.ts]: renamed to index

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

**Pull request merged**